File: examples/broadcast.c
#include <stdio.h>
#include <mpi.h>

int main (int argc, char * argv[]) {
    int rank, size, root = 0;
    float a[2];

    MPI_Init( &argc, &argv );
    MPI_Comm_rank( MPI_COMM_WORLD, &rank );
    MPI_Comm_size( MPI_COMM_WORLD, &size );

    if (rank == root) {
        a[0] = 2.0f;
        a[1] = 4.0f;
    }

    MPI_Bcast( a, 2, MPI_FLOAT, root,
               MPI_COMM_WORLD );
    printf( "%d: a[0]=%f\ta[1]=%f\n",
            rank, a[0], a[1] );

    MPI_Finalize();
    return 0;
}

File: examples/broadcast.f90
PROGRAM broadcast

USE mpi_f08
IMPLICIT NONE

INTEGER myid, nproc, root
TYPE(MPI_Status) :: status
REAL a(2)

    CALL MPI_Init()
    CALL MPI_Comm_size(MPI_COMM_WORLD, &
                       nproc)
    CALL MPI_Comm_rank(MPI_COMM_WORLD, &
                       myid)
    root = 0

    IF( myid == root ) THEN
        a(1) = 2.0
        a(2) = 4.0
    END IF

    CALL MPI_BCAST(a, 2, MPI_REAL, root, &
                   MPI_COMM_WORLD)
    WRITE(*,*) myid, ': a(1)=', a(1), &
                       'a(2)=', a(2)
    CALL MPI_Finalize()

END PROGRAM broadcast

